Content.Server/SimpleStation14/Weapons/Ranged/Systems/FireOnDropSystem.cs

```diff
@@ -0,0 +1,32 @@
+using Content.Shared.Throwing;
+using Content.Shared.Weapons.Ranged.Components;
+using Content.Shared.Weapons.Ranged.Systems;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Random;
+
+namespace Content.Server.SimpleStation14.Weapons.Ranged.Systems;
+
+public sealed class FireOnDropSystem : EntitySystem
+{
+    [Dependency] private readonly SharedGunSystem _gun = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GunComponent, LandEvent>(HandleLand);
+    }
+
+
+    private void HandleLand(EntityUid uid, GunComponent component, ref LandEvent args)
+    {
+        // TODO: This shouldn't be a hardcoded 10% roll. I wanted to base it off mass, but most items don't seem to care about their mass (most guns had the same).
+        // Then I wanted to base it off of item size, but the minigun still has a size of 5 at the time of writing, so ¯\_(ツ)_/¯
+        if (_random.Prob(0.1f))
+            _gun.AttemptShoot(uid, uid, component, Transform(uid).Coordinates.Offset(Transform(uid).LocalRotation.ToVec()));
```

Comment:
I was apparently wrong. This is actually failing because LandEvent is operating inside of an Enumeration over an object's physics component. Your code is running fine inside the enumeration, and it will even fire the weapon when it's thrown. However firing a weapon triggers a linearimpulse, which modifies the weapon's physics component, and since said physics component currently resides inside an enumeration, it causes the game to crash on the very next frame. What I think you might need to do is instead add an AfterLandEvent which runs outside of said enumeration. Essentially waiting until it has finished calculating physics for the object before asking it to calculate something new. This definitely isn't the best solution either. I can imagine some situations where you might say, "But I want the gun to be able to bounce off a wall, fire, and then change direction mid bounce because of the shot".

This is just a really odd coincedence it seems of how the game handles throwing an object.